pandas/tseries/holiday.py

```diff
@@ -406,17 +406,17 @@ def holidays(self, start=None, end=None, return_name=False):
         start = Timestamp(start)
         end = Timestamp(end)
 
-        holidays = None
+        holidays = []
         # If we don't have a cache or the dates are outside the prior cache, we
         # get them again
         if self._cache is None or start < self._cache[0] or end > self._cache[1]:
             for rule in self.rules:
-                rule_holidays = rule.dates(start, end, return_name=True)
+                holidays.append(rule.dates(start, end, return_name=True))
 
-                if holidays is None:
-                    holidays = rule_holidays
-                else:
-                    holidays = holidays.append(rule_holidays)
+            if holidays:
+                holidays = concat(holidays)
+            else:
+                holidays = Series(index=DatetimeIndex([]), dtype=object)
```

Comment:
Doesn't holidays typically return a DatetimeIndex ? Should maintain the same type even when empty

Reply:
Internally, the holidays are constructed with `return_name=True` which returns a `Series` with the holiday names as values:
```python
In [7]: USFederalHolidayCalendar().holidays('01-Jan-2020', '01-Apr-2020', return_name=True)
Out[7]: 
2020-01-01                 New Years Day
2020-01-20    Martin Luther King Jr. Day
2020-02-17                Presidents Day
dtype: object
```

Only if `return_name=False` are the values striped at the end of the method - behaviour which remains unchanged from master:
https://github.com/pandas-dev/pandas/blob/ba08390348403a0bddbfed835549982188e7038c/pandas/tseries/holiday.py#L426-L429